internal/app/cache/cache.go

```diff
@@ -49,7 +49,7 @@ type cache struct {
 
 type Resource struct {
 	Resp           transport.Response
-	Requests       map[transport.Request]bool
+	Requests       *sync.Map
```

Comment:
I feel that we should wrap the sync.Map implementation around a struct with accessors, similar to how the downstream orchestrator map is currently handling it. Reading/writing to interface can get hard to enforce.

Reply:
i've changed the sync.Map into a wrapper object to make it strongly typed.